### PR TITLE
Drop core-js in favour of custom fallbacks

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/atlassian/react-beautiful-dnd/issues/1629#issuecomment-572408426

I forbid using es6+ methods I added eslint-plugin-es6. It does not
forbid isInteger but I send PR https://github.com/nkt/eslint-plugin-es5/pull/37.

Looks like we saved 2kb gzipped. And one less commonjs dependency!

### CODE REVIEW

File: .size-snapshot.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-    "gzipped": 41966
+    "bundled": 365555,
+    "minified": 134065,
+    "gzipped": 39834
   },
   "dist/react-beautiful-dnd.min.js": {
-    "bundled": 321489,
-    "minified": 115006,
-    "gzipped": 34084
+    "bundled": 307334,
```
Comment (alexreardon) [inline .size-snapshot.json]: nice!

File: package.json [hunk 2]
```diff
@@ -56,7 +56,7 @@
     "prepublishOnly": "yarn build"
   },
   "dependencies": {
-    "@babel/runtime-corejs2": "^7.6.3",
+    "@babel/runtime": "^7.6.3",
```
Comment (alexreardon) [inline package.json]: thanks for sorting this!

File: src/view/assign-visually-hidden-style.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// @flow
+// https://allyjs.io/tutorials/hiding-elements.html
+// Element is visually hidden but is readable by screen readers
+const assignVisuallyHidden = (style: Object) => {
```
Comment (alexreardon) [inline src/view/assign-visually-hidden-style.js]: I prefer my functions not to mutate. Are we avoiding Object.assign?

Comment (TrySound) [reply src/view/assign-visually-hidden-style.js]: Yes. Object.assign need to be polyfilled. Though in this case much easier to get mutation. I think this is ok for this particular function. object assign polyfill would mutate too.

Comment (alexreardon) [reply src/view/assign-visually-hidden-style.js]: We use spread quite a bit in the library. Wouldn't that be using Object.assign?

Comment (TrySound) [reply src/view/assign-visually-hidden-style.js]: Spread uses extends helpers which comes from `@babel/runtime` package

Comment (alexreardon) [reply src/view/assign-visually-hidden-style.js]: But it won’t use that if you use object.assign. Can we import extends from Babel runtime and use that for the assign? 🤔

Comment (alexreardon) [reply src/view/assign-visually-hidden-style.js]: My ideal is to not have a function modify the passed in object but rather to return a new object (or to do the Object.assign in place) But I am okay to not get hung up on it

File: package.json [hunk 4]
```diff
@@ -70,6 +70,7 @@
     "@babel/core": "^7.6.4",
     "@babel/plugin-proposal-class-properties": "^7.5.5",
     "@babel/plugin-transform-modules-commonjs": "^7.6.0",
+    "@babel/plugin-transform-object-assign": "^7.8.3",
```
Comment (alexreardon) [inline package.json]: just to confirm - this won't polyfill, it is just ponyfill?

Comment (alexreardon) [reply package.json]: It looks fine: https://babeljs.io/docs/en/next/babel-plugin-transform-object-assign.html#example

Comment (TrySound) [reply package.json]: Yes

**Comment by TrySound** [comment]
You will be able to get rid from eslint suppression in new release of es5 plugin

**Comment by alexreardon** [comment]
Thanks for this @TrySound. I will aim to release this on monday

**Comment by davidchenfitzgerald** [comment]
Hey, when I re added react-beautiful-dnd, I still got the dependency in my lock file for corejs2. Did this merge completely remove it or is there something else I need to do?

**Comment by davidchenfitzgerald** [comment]
Are you guys still planning on a release with this soon?

**Comment by TrySound** [comment]
@davidchenfitzgerald Yes, after fixing a few minor issues

**Pull request merged**